hles/extraction/src/main/scala/org/broadinstitute/monster/dap/RedCapClient.scala

```diff
@@ -29,115 +26,100 @@ trait RedCapClient extends Serializable {
+          .add("exportCheckboxLabel", "false")
+          // Honestly not sure what these do, haven't seen the need to make them 'true'.
+          .add("exportSurveyFields", "false")
+          .add("exportDataAccessGroups", "false")
+        // Parameterized arm
+        if (generatorParams.arm.nonEmpty) {
+          formBuilder.add("events", generatorParams.arm.mkString(","))
+        }
+
+        ids.zipWithIndex.foreach {
+          case (id, i) => formBuilder.add(s"records[$i]", id)
+        }
+        fields.zipWithIndex.foreach {
+          case (f, i) => formBuilder.add(s"fields[$i]", f)
+        }
+        forms.zipWithIndex.foreach {
+          case (f, i) => formBuilder.add(s"forms[$i]", f)
+        }
+        start.foreach(s => formBuilder.add("dateRangeBegin", s.format(dateFormatter)))
+        end.foreach(e => formBuilder.add("dateRangeEnd", e.format(dateFormatter)))
+        if (filters.nonEmpty) {
+          formBuilder.add(
+            "filterLogic",
+            filters.map { directive =>
+              s"[${directive.field}]${directive.operation.op}${directive.comparand}"
+            }.mkString(" and ")
           )
-          logger.debug(s"Querying RedCap for data dictionary: ${logPieces.mkString(",")}")
-
-          new FormBody.Builder()
-            .add("token", apiToken)
-            // Export individual survey records as JSON.
-            .add("content", "metadata")
-            .add("format", "json")
-            .add("returnFormat", "json")
-            .add("forms[0]", instrument)
-      }
+        }
+        formBuilder
+      case GetDataDictionary(instrument) =>
+        val logPieces = List(
+          s"forms: [${instrument}]"
+        )
+        logger.debug(s"Querying RedCap for data dictionary: ${logPieces.mkString(",")}")
+
+        new FormBody.Builder()
+          .add("token", generatorParams.apiToken)
+          // Export individual survey records as JSON.
+          .add("content", "metadata")
+          .add("format", "json")
+          .add("returnFormat", "json")
+          .add("forms[0]", instrument)
+    }
 
-      val request = new Request.Builder()
-        .url(apiRoute)
-        .post(formBuilder.build())
-        .build()
+    new Request.Builder()
+      .url(apiRoute)
+      .post(formBuilder.build())
+      .build()
+  }
 
-      val p = Promise[Msg]()
-      client
-        .newCall(request)
-        .enqueue(new Callback {
-          override def onFailure(call: Call, e: IOException): Unit =
-            p.failure(e)
-          override def onResponse(call: Call, response: Response): Unit = {
-            val maybeResult =
-              JsonParser.parseEncodedJsonReturningFailure(response.body().string())
-            maybeResult match {
-              case Right(result) => p.success(result)
-              case Left(err)     => p.failure(err)
-            }
-          }
-        })
-      p.future
+  /** Construct a client instance backed by the production RedCap instance. */
```

Comment:
i don't know that this comment is accurate anymore, since the client construction happens in the `OkWrapper`.